### PR TITLE
General code cleanup of mapviz_plugins (Indigo)

### PR DESCRIPTION
**Pull request by pjreed**
This doesn't change any functionality; it's just cleaning up code.  Notably, this will:
- Fix all warnings (notably lots of ones about type casting)
- Move all .ui files to their own directory
- Remove unused variables
- Remove commented-out code
- Make spacing and indentation consistent
- Make brace style consistent

### CODE REVIEW

File: mapviz_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -120,6 +157,7 @@ install(TARGETS ${PROJECT_NAME}
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
 )
 
-install(FILES mapviz_plugins.xml
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+install(FILES ${PROJECT_NAME}.xml
```
Comment (evenator) [inline mapviz_plugins/CMakeLists.txt]: The fact that this file has the same name as the package is incidental; maybe it should be hardcoded.

**Comment by evenator** [comment]
I definitely support this. Remind me to test it--I want to run it a little before I merge it because it's too big to verify by inspection.

File: mapviz_plugins/CMakeLists.txt (hunk 1 above)
Comment (pjreed) [inline mapviz_plugins/CMakeLists.txt]: I don't have strong feelings either way, but `catkin_lint` complains if the CMake project name is used as a literal string anywhere in the file.

**Comment by pjreed** [comment]
Reminder for @evenator to test this some more.  Couldn't hurt for @malban to test it, too.

**Comment by evenator** [comment]
Unfortunately, I almost never run mapviz...

**Comment by malban** [comment]
Looks good to me.

**Comment by evenator** [comment]
At the risk of breaking everything, I recommend we go ahead and merge this.

**Pull request merged**